packages/ui-common/src/AlertsContext.tsx

```diff
@@ -0,0 +1,48 @@
+// Copyright 2018-2019 @paritytech/substrate-light-ui authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import React, { useState, createContext } from 'react';
+
+export type AlertType = 'error' | 'info' | 'success' | 'warning';
+
+export interface AlertWithoutId {
+  content: React.ReactNode;
+  type: AlertType;
+}
+
+export interface Alert extends AlertWithoutId {
+  id: number;
+}
+
+export const AlertsContext = createContext({
+  enqueue: (newAlertWithoutId: AlertWithoutId) => { console.error('No context provider found above in the tree.'); },
+  remove: (Alertid: number) => { console.error('No context provider found above in the tree.'); },
+  alerts: [] as Alert[] });
+
+interface Props {
+  children: any;
+}
+
+export function AlertsContextProvider (props: Props) {
+
+  const [{ alerts, nextAlertId }, setAlerts] = useState({ alerts: [] as Alert[], nextAlertId: 0 });
+
+  const enqueue = (newAlertWithoutId: AlertWithoutId) => {
```

Comment:
performance nitpick (however might not make a signficiant difference for now so I think we can ignore):
enqueue, dequeue are re-created on every render; this means that if `alerts` change, thus triggering a re-render of AlertsContextProvider, then enqueue and dequeue will be re-created, and as a result the effects only depending on enqueue inside components consuming AlertsContext will be retriggered, even though `enqueue` did not change, in practice.
any idea welcome!